update_annotations_from_table.py

```diff
@@ -0,0 +1,182 @@
+"""This notebook updates all file annotations a project.
+
+There are four required file annotations for any assay file uploaded to Synapse.
+
+`UID` - a universal ID for the sample represented in the file
+`dataType`: `mRNA`, `miRNA`, `methylation`, etc.
+`fileType`: `bam`, `fastq`, etc.
+`fileSubType`: is `fileType`-specific; e.g., `bam` can have `mapped` or `unmapped`.
+
+This script uses the `dataType` field to get assay-specific files (using a query lookup dictionary `assayToQuery`),
+and then the `UID` to merge existing file annotations and the table metadata (which is obtained through a lookup
+dictionary `assayToMetadataTable`.
+
+"""
+
+import sys
+import pandas as pd
+import numpy
```

Comment:
Pandas has <code>pandas.isnull()</code> for checking missing values (NaN in numeric arrays, None/NaN in object arrays).